shesmu-pluginapi/src/main/java/ca/on/oicr/gsi/shesmu/plugin/grouper/Grouper.java

```diff
@@ -0,0 +1,25 @@
+package ca.on.oicr.gsi.shesmu.plugin.grouper;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Perform a complex sub grouping operation
+ *
+ * @param <I> the input record type
+ * @param <O> the output record type
+ */
+public interface Grouper<I, O> {
+  /**
+   * Group the inputs provided into sensible subgroups. The grouper may drop input or assign the
+   * same input to multiple groups; it should not include the same input the same group multiple
+   * times
+   *
+   * @param inputs the inputs to be grouped; the list is now owned by the grouper and the grouper
```

Comment:
"inputs the inputs" "the inputs"

Reply:
The first `inputs` is part of the JavaDoc identifier. It doesn't highlight properly here.